src/index.ts

```diff
@@ -619,40 +619,65 @@ function negate(delimiter: string, backtrack: string) {
 }
 
 /**
- * Stringify token data into a path string.
+ * Stringify an array of tokens into a path string.
  */
-export function stringify(data: TokenData) {
-  return data.tokens
-    .map(function stringifyToken(token, index, tokens): string {
-      if (token.type === "text") return escapeText(token.value);
-      if (token.type === "group") {
-        return `{${token.tokens.map(stringifyToken).join("")}}`;
-      }
+function stringifyTokens(tokens: Token[]): string {
+  let value = "";
+  let i = 0;
+
+  function name(value: string) {
```

Comment:
Pulled `name` function out (which adds bytes) because it's needed to avoid a crash on `name` being missing _before_ the token type error.